### PR TITLE
[DOC] Fix more typos

### PR DESCRIPTION
**Pull request by szepeviktor**
#### Reference Issues/PRs

Follow-up #5424
Thank you @kianmeng

#### What does this implement/fix? Explain your changes.

Found few more misspellings.

### CODE REVIEW

**Comment by szepeviktor** [comment]
For `typos` to work it needs this configuration file.
`.typos.toml`
```toml
[files]
extend-exclude = [
    ".git/",
]
ignore-hidden = false

[default]
extend-ignore-re = [
    "Binay Kumar",
    "Edgaras and Mapp",
    'Sch"afer',
    "Sebastiaan Koel",
    "Xu, Ba,",
    "class 'sktime\\.\\S+\\.\\.\\.",
    "data format fo",
    "RotatIoN",
    "the BA matrix",
]

[default.extend-words]
"adaptee" = "adaptee"
"applicates" = "applicates"
"fpr" = "fpr"
"lik" = "lik"
"mape" = "mape"
"methodd" = "methodd"
"nd" = "nd"
"strat" = "strat"

[default.extend-identifiers]
"ip_ba" = "ip_ba"
"mp_ba" = "mp_ba"
```

File: sktime/performance_metrics/forecasting/_classes.py [hunk 1]
```diff
@@ -642,7 +642,7 @@ def custom_mae(y_true, y_pred) -> float:
 
             return float(result)
 
-        params2 = {"func": custom_mae, "name": "custom_nmae", "lower_is_better": True}
+        params2 = {"func": custom_mae, "name": "custom_mae", "lower_is_better": True}
```
Comment (szepeviktor) [inline sktime/performance_metrics/forecasting/_classes.py:645]: @fkiraly This is a hard error.

Comment (yarnabrina) [reply sktime/performance_metrics/forecasting/_classes.py:645]: This is undesirable for sure, and should be fixed, but why is this a "hard error"?

It's a text describing what's the metric supposed to measure, but what is the effect of having a wrong text in metric description that is only being used aas part of unit tests? Please let me know if I am missing something.

File: sktime/transformations/series/exponent.py [hunk 2]
```diff
@@ -189,7 +189,7 @@ def get_test_params(cls, parameter_set="default"):
 
 
 class SqrtTransformer(ExponentTransformer):
-    """Apply element-sise square root transformation to a time series.
+    """Apply element-size square root transformation to a time series.
```
Comment (yarnabrina) [inline sktime/transformations/series/exponent.py:192]:
```suggestion
    """Apply elementwise square root transformation to a time series.
```

This makes more sense in my opinion.

**Review by fkiraly — CHANGES_REQUESTED** [review]
Thanks!

Minor but blocking: there are some linting errors caused by changes in line length.

**Review by yarnabrina — CHANGES_REQUESTED** [review]
Thanks for your PR. I've made a clarificaition comment, and requested a typo fix.

I went through your suggestions regarding `typos`, how do you cfeel confident about the word mapping you created? For example, I see you have listed `strat`. How do we know for sure that all occurrences of `strat` is intended as an abbreviation of strategy, and not as an unintended typo for `start`?

If we add a configuration file like you suggested and use it as part of pre-commit configurations and run it in CI, is it a foolproof way? I asked @kianmeng earlier, but before we add such an automated spell checker in codebase, I feel we need to ensure no false positive, probably at a cost of few false negatives to avoid frequent CI failures. Let me know what you think.

Also, can you please run codespell/typos (with a word mapping of your choice) over the entire repository and see what all do you find, if that's not done already? If there's nothing new, you can consider adding those configurations as part of this PR itself, if other reviewers are okay with it.

File: sktime/transformations/series/exponent.py (hunk 2 above)
Comment (szepeviktor) [reply sktime/transformations/series/exponent.py]: Done.

File: sktime/performance_metrics/forecasting/_classes.py (hunk 1 above)
Comment (szepeviktor) [reply sktime/performance_metrics/forecasting/_classes.py:645]: I thought "name" is later processed.
This was a mistake.

**Comment by szepeviktor** [comment]
> Also, can you please run codespell/typos (with a word mapping of your choice) over the entire repository and see what all do you find, if that's not done already? If there's nothing new, you can consider adding those configurations as part of this PR itself, if other reviewers are okay with it.

I've run `typos` on the entire repository.
What should I do now?

**Comment by fkiraly** [comment]
> What should I do now?

Fix the linting errors :-)
Worst case, the code-quality CI failure logs on GitHub (click "details") tell you what to do.

Best case, you have linting set up locally and you see the issue already in pycharm/vscode/other
https://www.sktime.net/en/stable/developer_guide/coding_standards.html#setting-up-local-code-quality-checks

**Comment by szepeviktor** [comment]
All green 🍏

**Comment by fkiraly** [comment]
> I am working on a smart fridge

What's that, it heats when it's cold outside?

**Comment by fkiraly** [comment]
> you can consider adding those configurations as part of this PR itself, if other reviewers are okay with it.

@yarnabrina, I would actually prefer any changes to CI in separate PR. You know, in case we have to revert sth in the CI, then we wouldn't want to revert all the typo fixes too (or end up splitting things up manually).

**Comment by yarnabrina** [comment]
> > you can consider adding those configurations as part of this PR itself, if other reviewers are okay with it.
> 
> @yarnabrina, I would actually prefer any changes to CI in separate PR. You know, in case we have to revert sth in the CI, then we wouldn't want to revert all the typo fixes too (or end up splitting things up manually).

I thought that it'll be an addition to pre commit hooks, not a new CI file or etc. But separating in a new PR makes sense, let's do that.

**Review by fkiraly — APPROVED** [review]
My comments are addressed, thx

**Comment by szepeviktor** [comment]
@fkiraly Thank you for merging.
We are at near zero with the number of incorrect words.

**Comment by yarnabrina** [comment]
@szepeviktor @kianmeng I just ran `codespell sktime`,  
and got quite a few (229 according to `wc -l`) typos. Few are of course part of code, but quite a few are still left across python scripts and jupyter notebooks.

Can one of you please share your exact command and how are you executing? I wish to check if I'm running in some wrong way. I've installed v2.2.6 of codespell, if that matters.

**Comment by szepeviktor** [comment]
I do not use code spell as `typos` is a better tool.
https://github.com/sktime/sktime/pull/5478#issuecomment-1774201922

**Comment by kianmeng** [comment]
@yarnabrina See https://github.com/sktime/sktime/pull/5424

**Comment by yarnabrina** [comment]
@kianmeng I ran yesterday on main, after that PR is merged. So, it's not affecting here I believe. Let me know if I'm missing something. 

@szepeviktor I do get a lot with typos as well. I'll create a new issue with more details and share with you.

**Pull request merged**